float64 l_x
float64 l_y
float64 T
---